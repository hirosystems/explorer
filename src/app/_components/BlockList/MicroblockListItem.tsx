import React from 'react';
import { FiZap } from 'react-icons/fi';

import { Circle } from '../../../common/components/Circle';
import { MicroBlockLink } from '../../../common/components/ExplorerLinks';
import { toRelativeTime } from '../../../common/utils/utils';
import { Flex } from '../../../ui/Flex';
import { HStack } from '../../../ui/HStack';
import { Icon } from '../../../ui/Icon';
import { Stack } from '../../../ui/Stack';
import { Caption, Text, Title } from '../../../ui/typography';

export const MicroblockListItem: React.FC<{
  blockTime: number;
  hash: string;
  index: number;
}> = ({ blockTime, hash, index, ...rest }) => {
  return (
    <Flex
      justifyContent="space-between"
      py="24px"
      _hover={{
        borderLeftColor: 'accent',
      }}
      {...rest}
    >
      <HStack as="span" alignItems="center" gap={4} minWidth={0}>
        <Circle size="40px" flexShrink={0}>
          <Icon as={FiZap} size="16px" fill={'textCaption.light'} color={'textCaption.light'} />
        </Circle>
        <Stack gap={2} as="span" minWidth={0}>
          <Flex alignItems="center" minWidth={0}>
            <MicroBlockLink hash={hash}>
              <Title
                display="block"
                fontSize={'sm'}
                minWidth={0}
                textOverflow={'ellipsis'}
                overflow={'hidden'}
                whiteSpace={'nowrap'}
              >
                {hash}
              </Title>
            </MicroBlockLink>
          </Flex>
          <Caption display="block">Microblock</Caption>
        </Stack>
      </HStack>
      <Stack gap={2} textAlign="right" as="span" minWidth={'80px'}>
        <Text
          fontSize="14px"
          width="100%"
          textAlign="right"
          display="block"
          suppressHydrationWarning={true}
        >
          {toRelativeTime(blockTime * 1000)}
        </Text>
      </Stack>
    </Flex>
  );
};
