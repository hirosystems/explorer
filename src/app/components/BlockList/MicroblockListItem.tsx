import { toRelativeTime } from '@/common/utils';
import { MicroBlockLink } from '@/components/links';
import { Circle, Flex, Icon, Stack } from '@/ui/components';
import { Caption, Text, Title } from '@/ui/typography';
import { useColorMode } from '@chakra-ui/react';
import React from 'react';
import { FiZap } from 'react-icons/fi';

export const MicroblockListItem: React.FC<{
  blockTime: number;
  hash: string;
  index: number;
  length: number;
}> = ({ blockTime, hash, index, length, ...rest }) => {
  return (
    <Flex
      justifyContent="space-between"
      py="24px"
      color={'textBody'}
      _hover={{
        borderLeftColor: 'accent',
      }}
      {...rest}
    >
      <Stack as="span" isInline alignItems="center" spacing="16px" minWidth={0}>
        <Circle size="40px" flexShrink={0}>
          <Icon as={FiZap} size="16px" fill={'textCaption.light'} color={'textCaption.light'} />
        </Circle>
        <Stack spacing="8px" as="span" minWidth={0}>
          <Flex alignItems="center" minWidth={0}>
            <MicroBlockLink hash={hash}>
              <Title
                display="block"
                fontSize={'14px'}
                as="a"
                color={`links.${useColorMode().colorMode}`}
                _hover={{
                  cursor: 'pointer',
                  textDecoration: 'underline',
                }}
                minWidth={0}
                textOverflow={'ellipsis'}
                overflow={'hidden'}
                whiteSpace={'nowrap'}
              >
                {hash}
              </Title>
            </MicroBlockLink>
          </Flex>
          <Caption display="block">Microblock</Caption>
        </Stack>
      </Stack>
      <Stack spacing="8px" textAlign="right" as="span" minWidth={'80px'}>
        <Text
          fontSize="14px"
          width="100%"
          textAlign="right"
          display="block"
          suppressHydrationWarning={true}
        >
          {toRelativeTime(blockTime * 1000)}
        </Text>
      </Stack>
    </Flex>
  );
};
