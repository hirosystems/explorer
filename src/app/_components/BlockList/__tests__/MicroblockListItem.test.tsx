import { render } from '@testing-library/react';
import React from 'react';

import { MicroblockListItem } from '../MicroblockListItem';

jest.mock('../../../../common/utils/utils', () => ({
  toRelativeTime: jest.fn(time => `MockRelativeTime${time}`),
}));

describe('MicroblockListItem component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <MicroblockListItem blockTime={1619047871} hash="mockHash" index={1} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
