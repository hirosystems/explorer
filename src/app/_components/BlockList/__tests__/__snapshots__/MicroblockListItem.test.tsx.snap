// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MicroblockListItem component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1qnwiy1"
  >
    <span
      class="chakra-stack css-19ovbli"
    >
      <div
        class="css-197es7z"
      >
        <svg
          class="chakra-icon css-1bc6rbx"
          fill="none"
          focusable="false"
          height="16px"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="13 2 3 14 12 14 11 22 21 10 12 10 13 2"
          />
        </svg>
      </div>
      <span
        class="chakra-stack css-on31qd"
      >
        <div
          class="css-ycwyyy"
        >
          <a
            class="chakra-link css-0"
            href="/microblock/mockHash?chain=mainnet"
          >
            <span
              class="chakra-text css-10wgqlb"
            >
              mockHash
            </span>
          </a>
        </div>
        <span
          class="chakra-text css-15iimks"
          style="user-select: none;"
        >
          Microblock
        </span>
      </span>
    </span>
    <span
      class="chakra-stack css-fdzsh0"
    >
      <span
        class="chakra-text css-ui7t6h"
      >
        MockRelativeTime1619047871000
      </span>
    </span>
  </div>
</DocumentFragment>
`;
